tests/lib/ocpp/v201/test_charge_point.cpp

```diff
@@ -340,4 +534,92 @@ TEST_F(ChargePointFixture, K01FR02_CallbacksValidityChecksIfOptionalTransactionE
     EXPECT_TRUE(callbacks.all_callbacks_valid());
 }
 
+TEST_F(ChargePointFixture, K01_SetChargingProfileRequest_ValidatesProfile) {
+    auto periods = create_charging_schedule_periods({0, 1, 2});
+
+    auto profile = create_charging_profile(
+        DEFAULT_PROFILE_ID, ChargingProfilePurposeEnum::TxProfile,
+        create_charge_schedule(ChargingRateUnitEnum::A, periods, ocpp::DateTime("2024-01-17T17:00:00")), DEFAULT_TX_ID);
+
+    SetChargingProfileRequest req;
+    req.evseId = DEFAULT_EVSE_ID;
+    req.chargingProfile = profile;
+
+    auto set_charging_profile_req =
+        request_to_enhanced_message<SetChargingProfileRequest, MessageType::SetChargingProfile>(req);
+
+    EXPECT_CALL(*smart_charging_handler, validate_profile(testing::_, testing::_));
```

Comment:
We know what will be passed into validate_profile(), right? Can we validate that it is called with those arguments rather than use teting::_ ?

Reply:
Yes and no; We need to figure out where to put the equality operator implementation so that both this and the other tests can use it.

Reply:
Christoph has a set of utils in his composite schedule work, so I didn't want to clash with it. That's why it's `testing::_` for now.